op-bindings/bindgen/main.go

```diff
@@ -0,0 +1,250 @@
+		gethLog.Crit("Error generating local bindings", "error", err.Error())
+	}
+	return nil
+}
+
+func generateRemoteBindings(c *cli.Context) error {
+	if err := configureLogger(c); err != nil {
+		gethLog.Crit("Error configuring logger", "error", err.Error())
+	}
+
+	if c.Bool("compare-deployment-bytecode") || c.Bool("compare-init-bytecode") {
+		if c.String("compare-chainid") == "" {
+			gethLog.Crit("In order to compare the bytecode against another chain, compare-chainid must be provided")
+
+		}
+
+		if c.String("compare-apikey") == "" {
+			gethLog.Crit("In order to compare the bytecode against another chain, compare-apikey must be provided")
+
+		}
+	}
+
+	var client contractDataClient
+	switch c.String("client") {
+	case "etherscan":
+		var err error
+		client, err = etherscan.NewClient(
+			c.Int("source-chainid"),
+			c.Int("compare-chainid"),
+			c.String("source-apikey"),
+			c.String("compare-apikey"),
+			c.Int("api-max-retries"),
+			c.Int("api-retry-delay"),
+		)
+		if err != nil {
+			gethLog.Crit("Error initializing new Etherscan client", "error", err.Error())
+
+		}
+	default:
+		gethLog.Crit(fmt.Sprintf("Unsupported client provided: %s", c.String("client")))
+	}
+
+	bindgen := NewRemoteBindingsGenerator(
+		c.String("remote-contracts"),
+		c.String("metadata-out"),
+		c.String("go-package"),
+		client,
+		c.Bool("compare-deployment-bytecode"),
+		c.Bool("compare-init-bytecode"),
+		c.Int("source-chainid"),
+		c.Int("compare-chainid"),
+	)
+	if err := bindgen.genBindings(); err != nil {
+		gethLog.Crit("Error generating remote bindings", "error", err.Error())
+	}
+	return nil
+}
+
+func generateFlags() []cli.Flag {
+	return []cli.Flag{
+		&cli.StringFlag{
+			Name:     "metadata-out",
+			Usage:    "Output directory to put contract metadata files in",
+			Required: true,
+		},
+		&cli.StringFlag{
+			Name:     "go-package",
+			Usage:    "Go package name given to generated files",
+			Required: true,
+		},
+		&cli.StringFlag{
+			Name:     "monorepo-base",
+			Usage:    "Path to the base of the monorepo",
+			Required: true,
+		},
+		&cli.StringFlag{
+			Name:     "log-level",
+			Usage:    "Configured the level of logs that will be printed, defaulted to info. Options: none, debug, info, warn, error, crit",
+			Required: false,
+		},
+	}
+}
+
+func localFlags() []cli.Flag {
+	return []cli.Flag{
+		&cli.StringFlag{
+			Name:     "local-contracts",
+			Usage:    "Path to file containing list of contracts to generate bindings for that have Forge artifacts available locally",
+			Required: true,
+		},
+		&cli.StringFlag{
+			Name:  "source-maps-list",
+			Usage: "Comma-separated list of contracts to generate source-maps for",
+		},
+		&cli.StringFlag{
+			Name:     "forge-artifacts",
+			Usage:    "Path to forge-artifacts directory, containing compiled contract artifacts",
+			Required: true,
+		},
+	}
+}
+
+func remoteFlags() []cli.Flag {
+	return []cli.Flag{
+		&cli.StringFlag{
+			Name:     "client",
```

Comment:
i would also add `etherscan-*` as a prefix to all flags specific to etherscan like the api key

Reply:
One of the reasons for this is because many of the OP chains don't have an Etherscan, but instead use Blockscout. For MVP, we don't need to worry about implementing another client, but the code at least provides some direction for someone to extend it to support another client when needed